Docs processing - don't break markdown input

mpajkowski opened the pull request:
This PR makes Markdown parsers happier by disabling line merging.

ralpha commented:
Note: There has already been some discussion about this some time ago.
See #120 
So make sure this PR works for everyone.

mpajkowski commented:
It's a breaking change for users relying on current behaviour, indeed. 

I think I can add flag to `SchemaSettings` which enables/disables stripping exceeding lines (on by default).
I considered custom de-stripping `Visitor` for a moment, but I think it will be easier to keep it raw at this stage.

mpajkowski commented:
I reworked my solution.

I created `SchemaSettings::raw_description_text` flag which toggles described behaviour (set to `false` by default). Actually this flag is honoured on the _derive_ side - This way we only touch doc comments generated by `JsonSchema` macro.

I also updated tests and docs.
